mutex_linux.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+// +build linux
+
+package mutex
+
+import (
+	"net"
+	"path/filepath"
+
+	"github.com/juju/errors"
+)
+
+const prefix = "@/var/lib/juju/mutex-"
+
+type mutex struct {
+	socket *net.UnixListener
+}
+
+func acquire(name string) (Releaser, error) {
+	path := filepath.Join(prefix, name)
+	addr, err := net.ResolveUnixAddr("unix", path)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	l, err := net.ListenUnix("unix", addr)
+	if err != nil {
+		return nil, errLocked
+	}
+	return &mutex{socket: l}, nil
+}
+
+// Release implements Releaser.
+func (m *mutex) Release() {
+	if m.socket == nil {
```

Comment:
Do you support this ?
```go
r := mutex.Acquire()
go r.Release()
r.Release()
```
You may need to add a lock to prevent two Release's happening concurrently as m.socket is not nil'd out until after the call to close.